.github/workflows/rust.yml

```diff
@@ -96,7 +96,30 @@ jobs:
       - name: Check clippy
         working-directory: ${{ matrix.directory }}
         if: always()
-        run: cargo +${{ matrix.toolchain }} make clippy -- -D warnings
+        run: |
+          cargo +${{ matrix.toolchain }} make clippy --message-format=json \
+            | clippy-sarif \
+            | jq '.runs[].results |= unique' \
```

Comment:
With https://github.com/psastras/sarif-rs/pull/429 (hopefully) landing soon, what would happen if `relatedLocations` is different but the error is the same? I imagine in that case, we would emit the error twice.

This scenario is more likely than one might imagine, e.g., if we have conditional code that emits a related error.

Reply:
How about leveraging `reduce` and merging if they have the same `ruleId`?

I think even better would be the use of [`@microsoft/sarif-multitool`](https://github.com/microsoft/sarif-sdk/blob/main/docs/multitool-usage.md) which already has a merge operation.

Reply:
If the related locations are different it's a different error. This is purely to deduplicate errors that are exactly the same because e.g. clippy runs with a different permutation of features. I don't see how the tool you linked would help us here. The only real use case I see is adjusting the file paths, but it's not worth it for a one-liner to be replaced with a tool.